databaseutil/databaseutil.go

```diff
@@ -3,19 +3,73 @@ package databaseutil
 import (
 	"database/sql"
 	// Notice that we’re loading the driver anonymously, The driver registers itself as being available to the database/sql package.
+	"fmt"
 	_ "github.com/lib/pq"
+	"golang.org/x/crypto/bcrypt"
+	"time"
 )
 
-func Connect(dbUrl string) (*sql.DB, error) {
-	db, err := sql.Open("postgres", dbUrl)
+var db *sql.DB
+
+func Connect(dbUrl string) error {
+	temp, err := sql.Open("postgres", dbUrl)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
+	db = temp
+
 	// Quickly test if the connection to the database worked.
 	if err := db.Ping(); err != nil {
-		return nil, err
+		return err
+	}
+
+	return nil
+}
+
+func SaveNewUser(
+	displayName string,
+	emailAddress string,
+	password string) (int64, error) {
+
+	hash, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		return -1, err
+	}
+
+	sqlStatement := `
+		INSERT INTO users (display_name, email_address, password, creation_time) 
+		VALUES ($1, $2, $3, $4) RETURNING id`
+
+	var id int64
+	err = db.QueryRow(sqlStatement, displayName, emailAddress, hash, time.Now().UTC()).Scan(&id)
+	if err != nil {
+		return -1, err
+	}
+
+	fmt.Sprintf("%d", id)
```

Comment:
we can leave this here for now, but we need a way to remember to clean up our prints when they're no longer useful